app/scripts/controllers/account-tracker-controller.test.ts

```diff
@@ -767,6 +767,55 @@ describe('AccountTrackerController', () => {
     });
   });
 
+  describe('updateAccountByAddress', () => {
+    it('does not update account by address if completedOnboarding is false', async () => {
+      await withController(
+        {
+          completedOnboarding: false,
+        },
+        async ({ controller }) => {
+          const VALID_ADDRESS_TO_UPDATE = '0x1234';
+          await controller.updateAccountByAddress({
+            address: VALID_ADDRESS_TO_UPDATE,
+          });
+
+          expect(controller.state).toStrictEqual({
+            accounts: {},
+            currentBlockGasLimit: '',
+            accountsByChainId: {},
+            currentBlockGasLimitByChainId: {},
+          });
+        },
+      );
+    });
+
+    it('updates an account by address if completedOnboarding is true', async () => {
```

Comment:
Awesome tests!

Should we add 1 more test for when you don't provide an address property (so it uses the selected account)?